src/explorer/SubscriptionTreeItem.ts

```diff
@@ -105,9 +106,41 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
             const location: Location = nonNullProp(wizardContext, 'location');
             wizardContext.newResourceGroupName = `appsvc_rg_${wizardContext.newSiteOS}_${location.name}`;
             wizardContext.newPlanName = `appsvc_asp_${wizardContext.newSiteOS}_${location.name}`;
+
+            const asp: AppServicePlan | null = await getAppServicePlan(wizardContext, wizardContext.newResourceGroupName, wizardContext.newPlanName);
+            if (asp && checkPlanForPerformanceDrop(asp)) {
+                // Subscriptions can only have 1 free tier Linux plan so show the warning if there are too many apps on the plan
+
+                if (isPlanLinux(asp)) {
+                    const showPerfWarning: boolean | undefined = workspaceConfig.get(configurationSettings.showPlanPerformanceWarning);
```

Comment:
I would prefer if you moved this logic inside of `showPerformancePrompt` (You'd probably have to rename the function at that point)

Otherwise people might look at `showPerformancePrompt` and think you forget to check the "Dont warn again" setting. Also then you don't have to define `showPlanPerformanceWarning` in `constants.ts` since it'll only be used in one function.